src/main/kotlin/calculator/Operation.kt

```diff
@@ -0,0 +1,44 @@
+package calculator
+
+enum class Operation(val symbol: String) {
+    ADD("+") {
+        override fun apply(
+            a: Int,
+            b: Int,
+        ) = a + b
+    },
+    SUBTRACT("-") {
+        override fun apply(
+            a: Int,
+            b: Int,
+        ) = a - b
+    },
+    MULTIPLY("*") {
+        override fun apply(
+            a: Int,
+            b: Int,
+        ) = a * b
+    },
+    DIVIDE("/") {
+        override fun apply(
+            a: Int,
+            b: Int,
+        ): Int {
+            require(b != 0) { "0으로 나눌 수 없습니다." }
+            return a / b
+        }
+    },
+    ;
+
+    abstract fun apply(
+        a: Int,
+        b: Int,
+    ): Int
```

Comment:
operator를 enum class로 잘 관리해주셨군요? 👍💯 

[SAM](https://kotlinlang.org/docs/fun-interfaces.html)을 활용하여 아래와 같이 변경해볼 수도 있을 것 같아요. 😃

```suggestion
enum class Operation(val symbol: String, operate: (Int, Int) -> Int) {
    ADD("+", { a, b -> a + b }),
    SUBTRACT("-", { a, b -> a - b }),
    MULTIPLY("*", { a, b -> a * b }),
    DIVIDE("/", { a, b ->
        require(b != 0) { "0으로 나눌 수 없습니다." }
        a / b
    })
    ;

    fun apply(a: Int, b: Int): Int = this.operate(a, b)
```